java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -575,7 +618,7 @@ static ValueRange getValueRange(ColumnStatistics index,
       Boolean max = stats.getTrueCount() != 0;
       return new ValueRange<>(predicate, min, max, stats.hasNull());
     } else {
-      return new ValueRange(predicate, null, null, true);
+      return new ValueRange(predicate, index.hasNull(), true, false);
```

Comment:
This change looks a little misleading. Could you use the full constructor?

Reply:
This is a great suggestion, already using the full constructor